File: backend/src/main/java/com/example/service/ReclamationService.java
package com.example.service;

import com.example.model.Reclamation;

import java.util.List;

public interface ReclamationService {

    public List<Reclamation> getAllReclamations();

    public Reclamation create (Reclamation reclamation);

    public void delete(Long id);

}

File: frontend/src/app/services/reclamation.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {User} from "../pages/models/user";

@Injectable({
  providedIn: 'root'
})
export class ReclamationService {
reclamationURL = 'http://localhost:8080/api/reclamation';

  constructor(private httpClient: HttpClient) { }

  createReclamation(data: any) {
    return this.httpClient.post(this.reclamationURL + '/utilisateur/reclamation',data);
  }
  getAllReclamations() {
    return this.httpClient.get(this.reclamationURL);
  }
  deleteReclamation(id: any) {
    return this.httpClient.delete(`${this.reclamationURL}/${id}`);
  }
}
